NCS.DSS.Interaction/PostInteractionHttpTrigger/Function/PostInteractionHttpTrigger.cs

```diff
@@ -44,70 +44,105 @@ public PostInteractionHttpTrigger(IResourceHelper resourceHelper, IHttpRequestHe
+
             var touchpointId = _httpRequestMessageHelper.GetDssTouchpointId(req);
             if (string.IsNullOrEmpty(touchpointId))
             {
-                log.LogInformation("Unable to locate 'APIM-TouchpointId' in request header.");
+                _logger.LogInformation("Unable to locate 'TouchpointId' in request header.");
                 return new BadRequestObjectResult(HttpStatusCode.BadRequest);
             }
 
             var ApimURL = _httpRequestMessageHelper.GetDssApimUrl(req);
             if (string.IsNullOrEmpty(ApimURL))
             {
-                log.LogInformation("Unable to locate 'apimurl' in request header");
+                _logger.LogInformation("Unable to locate 'apimurl' in request header");
                 return new BadRequestObjectResult(HttpStatusCode.BadRequest);
             }
 
-            log.LogInformation("Post Interaction C# HTTP trigger function processed a request. " + touchpointId);
+            _logger.LogInformation("Post Interaction C# HTTP trigger function processed a request. " + touchpointId);
 
             if (!Guid.TryParse(customerId, out var customerGuid))
+            {
+                _logger.LogWarning("Unable to parse 'customerId' to a GUID. Customer GUID: {CustomerID}", customerId);
                 return new BadRequestObjectResult(customerGuid);
+            }
 
+            _logger.LogInformation("Attempting to retrieve resource from request.");
             Models.Interaction interactionRequest;
-
             try
             {
                 interactionRequest = await _httpRequestMessageHelper.GetResourceFromRequest<Models.Interaction>(req);
             }
             catch (Exception ex)
             {
+                _logger.LogError(ex, "Unable to parse {interactionRequest} from request body. Exception: {ExceptionMessage}", nameof(interactionRequest), ex.Message);
                 return new UnprocessableEntityObjectResult(_dynamicHelper.ExcludeProperty(ex, ["TargetSite"]));
             }
 
             if (interactionRequest == null)
+            {
+                _logger.LogWarning("{interactionRequest} object is NULL.", nameof(interactionRequest));
                 return new UnprocessableEntityObjectResult(req);
+            }
 
             interactionRequest.SetIds(customerGuid, touchpointId);
 
+            _logger.LogInformation("Attempting to validate {interactionRequest} object", nameof(interactionRequest));
             var errors = _validate.ValidateResource(interactionRequest);
-
             if (errors != null && errors.Any())
+            {
+                _logger.LogWarning("Falied to validate {interactionRequest} object", nameof(interactionRequest));
                 return new UnprocessableEntityObjectResult(errors);
+            }
+            _logger.LogInformation("Successfully validated {interactionRequest} object", nameof(interactionRequest));
 
```

Comment:
Need a log stating that input validation is complete, see https://github.com/SkillsFundingAgency/dss-transfer/pull/31/files#diff-695bbf80412b14a3c3dcfde0ff6fd17ff9d7119b08dcd1741520ff6e7ee8cd9f line 71 for an example